delocate/tools.py

```diff
@@ -393,14 +411,19 @@ def find_package_dirs(root_path):
     Returns
     -------
     package_sdirs : set
-        Set of strings where each is a subdirectory of `root_path`, containing
-        an ``__init__.py`` file.  Paths prefixed by `root_path`
+        Set of (string, is_dir) tuples where each string is a subdirectory of
+        `root_path` containing an ``__init__.py`` file, or a C extension
+        single-file module directly in `root_path`, and is_dir indicates whether
+        the package is a subdirectory module or a single-file module.  Paths
+        prefixed by `root_path`
     """
     package_sdirs = set()
     for entry in os.listdir(root_path):
         fname = entry if root_path == '.' else pjoin(root_path, entry)
         if isdir(fname) and exists(pjoin(fname, '__init__.py')):
-            package_sdirs.add(fname)
+            package_sdirs.add((fname, True))
+        elif isfile(fname) and dylibs_only(fname):
```

Comment:
Here I was thinking you'd use lib_filt_func ...

Reply:
What should the behavior be if `lib_filt_func` is `None`, as is the default?

If I do:

```python
elif isfile(fname) and lib_filt_func and lib_filt_func(fname):
```

Then `delocate-wheel -d` is required in order to properly delocate top level C extension packages. Otherwise the behavior is the same as delocate currently works: top level C extension packages are included but not delocated.

If instead I do:

```python
elif isfile(fname) and ((lib_filt_func and lib_filt_func(fname)) or (lib_filt_func is None and dylibs_only(fname))):
```

Then it works without `-d`, but this seems against your intent as to the purpose of `lib_filt_func`?

Is the idea that `-d` should be required in order to delocate top level C extension packages? Or if `-d` is not used, should all files in the root be inspected, and in both cases, a more sophisticated method than `dylibs_only()` (i.e. binary inspection) should be used to determine if the file is a C extension package?

Reply:
Yes, sorry, you're right, I forgot that `dylibs_only` is not the default.    I'll merge this manually, and do a few edits to the docstrings to remind myself what is going on.